### ISSUE TITLE
Finetune on outputs only

### ISSUE THREAD

**Issue by louisoutin** [OptimalScale/LMFlow#551]
Hello, do you have an option to disable backpropagation for the question part?
(as the 'train_on_inputs' argument on alpaca-lora project: https://github.com/tloen/alpaca-lora/blob/main/finetune.py#L49)
I didn't find this option in lmflow codebase. For some cases, it might converge faster to only backpropagate on answers parts only.
Also, I'm seeing the group_texts method in the finetuner class: https://github.com/OptimalScale/LMFlow/blob/main/src/lmflow/pipeline/finetuner.py#L112 .  I assume batch_size = 1 during the finetuning process then? Is that correct?
Thanks.

**Comment by shizhediao** [comment]
Hi,
if you want to use the gradients from the answer part only, you may use our `text2text` format instead of `text_only` format. Please refer to https://optimalscale.github.io/LMFlow/examples/DATASETS.html

Thanks

**Comment by nliu86** [comment]
Do you have any working examples of training using text2text format? I tried but it doesn't work.

**Comment by shizhediao** [comment]
Hi,
What is the issue you observed when running with `text2text`?

**Comment by nliu86** [comment]
training with text2text actually works after I tried different parameters. Maybe it failed initially because I set block_size to 1.
